common/OpTestHMC.py

```diff
@@ -59,11 +59,28 @@ class OpSecureBootUtilities():
     '''
     This class addresses the secureboot related pre-requisites.
     '''
-
-    def __init__(self, console=None, distro=None):
+    def __init__(self, cv_hmc=None, hmc_con=None, console=None, distro=None):
+        self.cv_hmc = cv_hmc
+        self.hmc_con = hmc_con
         self.console = console
         self.distro = distro
 
+    def check_hmc_level_secureboot_state(self):
+        '''
+        Check whether the secure-boot is enabled and HMC level.
+        Return:
+        'True' in case of Secure boot enabled
+        'False' in case of Secure boot disabled
+        '''
+        # HMC command to know the current state of Secure Boot
+        cmd = ("lssyscfg -r lpar -m %s -F curr_secure_boot --filter lpar_names=%s" 
+               % (self.cv_hmc.mg_system, self.cv_hmc.lpar_name))
+        output = self.hmc_con.run_command(cmd, timeout=300)
+        if int(output[0]) == 2: # Value '2' means Secure Boot enabled
+            return True
+        elif int(output[0]) == 0: # Value '0' means Secure Boot disabled
```

Comment:
is this check required, is there any other states other then enalbed or disabled.

Reply:
Yes, as of now only two states exist. However, there could be a possibility of one more state in the future. Hence the conditions.